src/cavendish_particle_tracks/_decay_angles_dialog.py

```diff
@@ -55,30 +53,83 @@ def __init__(self, parent=None):
         ):
             self.layout().addWidget(table_datum, i % 3 + 2, i // 3)
 
-        self.layout().addWidget(bss, 4, 0, 1, 3)
-
+        self.layout().addWidget(btn_calculate, 5, 0, 1, 3)
         self.layout().addWidget(
             QLabel("Opening angles"),
-            5,
+            6,
             0,
             1,
             3,
         )
         for i, widget in enumerate(
             [QLabel("ϕ_proton [rad]"), QLabel("ϕ_pion [rad]")] + self.textboxes_phi
         ):
-            self.layout().addWidget(widget, i % 2 + 6, i // 2 + 1)
-        self.layout().addWidget(bap, 8, 0, 1, 3)
-        self.layout().addWidget(self.buttonBox, 10, 0, 1, 3)
-
+            self.layout().addWidget(widget, i % 2 + 7, i // 2 + 1)
+        self.layout().addWidget(btn_save, 9, 0, 1, 3)
+        self.layout().addWidget(self.buttonBox, 11, 0, 1, 3)
+        # endregion
         # Setup shapes layer
-        self.cal_layer = self._setup_decayangles_layer()
+        self.join_coordinates = [200, 300]
+        self.cal_layer: Shapes = self._setup_decayangles_layer()
+        self.cal_layer.events.data.connect(self._enforce_points_coincident)
+
+        # # This almost gets the behavior we want, but only for the line labels
+        # def enforce_points_coincident_v2(event: Event) -> None:
+        #     (selected_shape,) = self.cal_layer.selected_data
+        #     data = self.cal_layer.data
+        #     if len(data) != 3:
+        #         return
+        #     if selected_shape==0 and (data[0][0] != data[1][0]).all():
+        #         # # This is what we want but needs private methods
+        #         # vertices = self.cal_layer._data_view.shapes[1].data
+        #         # vertices[0] = data[0][0]
+        #         # self.cal_layer._data_view.edit(1, vertices, new_type=None)
+        #         # self.cal_layer.refresh()
+        #         data[1][0][0] = data[0][0][0]
+        #         data[1][0][1] = data[0][0][1]
+        #         self.cal_layer.refresh()
+        # self.cal_layer.events.set_data.connect(enforce_points_coincident_v2)
```

Comment:
```suggestion
```
Don't want commented-out code in production.

Reply:
OK, I'll open another PR.